airbyte-workers/src/main/java/io/airbyte/workers/temporal/scheduling/ConnectionManagerWorkflowImpl.java

```diff
@@ -398,4 +333,139 @@ private void continueAsNew(final ConnectionUpdaterInput connectionUpdaterInput)
+
+    final ScheduleRetrieverOutput scheduleRetrieverOutput = runMandatoryActivityWithOutput(configFetchActivity::getTimeToWait,
+        scheduleRetrieverInput);
+
+    return scheduleRetrieverOutput.getTimeToWait();
+  }
+
+  /**
+   * Creates a new job if it is not present in the input. If the jobId is specified in the input of
+   * the connectionManagerWorkflow, we will return it. Otherwise we will create a job and return its
+   * id.
+   */
+  private Long getOrCreateJobId(final ConnectionUpdaterInput connectionUpdaterInput) {
+    if (connectionUpdaterInput.getJobId() != null) {
+      return connectionUpdaterInput.getJobId();
+    }
+
+    final JobCreationOutput jobCreationOutput =
+        runMandatoryActivityWithOutput(
+            jobCreationAndStatusUpdateActivity::createNewJob,
+            new JobCreationInput(
+                connectionUpdaterInput.getConnectionId(), workflowState.isResetConnection()));
+    connectionUpdaterInput.setJobId(jobCreationOutput.getJobId());
+
+    return jobCreationOutput.getJobId();
+  }
+
+  /**
+   * Create a new attempt for a given jobId
+   */
+  private Integer createAttemptId(long jobId) {
+    final AttemptCreationOutput attemptCreationOutput =
+        runMandatoryActivityWithOutput(
+            jobCreationAndStatusUpdateActivity::createNewAttempt,
+            new AttemptCreationInput(
+                jobId));
+
+    return attemptCreationOutput.getAttemptId();
+  }
+
+  /**
+   * Generate the input that is needed by the job. It will generate the configuration needed by the
+   * job and will generate a different output if the job is a sync or a reset.
+   */
+  private GeneratedJobInput getJobInput() {
+    final SyncInput getSyncInputActivitySyncInput = new SyncInput(
+        attemptId,
+        jobId,
+        workflowState.isResetConnection());
+
+    final GeneratedJobInput syncWorkflowInputs = runMandatoryActivityWithOutput(
+        getSyncInputActivity::getSyncWorkflowInput,
+        getSyncInputActivitySyncInput);
+
+    return syncWorkflowInputs;
+  }
+
+  /**
+   * Report the job as started in the job tracker and set it as running in the workflow internal
+   * state.
+   */
+  private void reportJobStarting() {
+    runMandatoryActivity(
+        jobCreationAndStatusUpdateActivity::reportJobStart,
+        new ReportJobStartInput(
+            jobId));
+
+    workflowState.setRunning(true);
+  }
+
+  /**
+   * Start the child SyncWorkflow
+   */
+  private StandardSyncOutput runChildWorkflow(GeneratedJobInput jobInputs) {
+    final SyncWorkflow childSync = Workflow.newChildWorkflowStub(SyncWorkflow.class,
+        ChildWorkflowOptions.newBuilder()
+            .setWorkflowId("sync_" + jobId)
+            .setTaskQueue(TemporalJobType.CONNECTION_UPDATER.name())
+            // This will cancel the child workflow when the parent is terminated
+            .setParentClosePolicy(ParentClosePolicy.PARENT_CLOSE_POLICY_TERMINATE)
```

Comment:
Going to be some merge conflicts here. Will need to make sure this is updated correctly in it.

Reply:
Yep. I have seen your review, the resolution should be easy